google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -93,6 +93,19 @@ void ToJson(std::chrono::system_clock::time_point const& field,
   j[name] = std::to_string(m);
 }
 
+void removeEmptyArraysAndObjects(nlohmann::json& j) {
+  for (auto item = j.begin(); item != j.end(); ) {
+    if ((item.value().is_array() || item.value().is_object()) && !item.value().empty()) {
+      removeEmptyArraysAndObjects(*item);
+    }
+    if (item.value().empty()) {
+      j.erase(item++);
```

Comment:
The documentation says:

```
Invalidates iterators and references at or after the point of the erase, including the end() iterator.
```

That suggests you cannot use `item` after calling `.erase(item++)`.

Did I miss something?

Reply:
I found such example here
https://github.com/nlohmann/json/issues/884#issuecomment-352977847
Quite old, but I couldn't find anything better.
Also documentation says:
```
Return value
Iterator following the last removed element. If the iterator pos refers to the last element, the end() iterator is returned.
```
Also I added another value in tests after empty array/object. Works ok.